lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -350,29 +361,30 @@ def main():
 
         # If VPC configuration is desired
         if vpc_subnet_ids or vpc_security_group_ids:
-            if len(vpc_subnet_ids) < 1:
-                module.fail_json(msg='At least 1 subnet is required')
-
-            if len(vpc_security_group_ids) < 1:
-                module.fail_json(msg='At least 1 security group is required')
+            try:
+                assert len(vpc_subnet_ids) > 0
```

Comment:
These asserts may be optimized out, so you shouldn't be using assertions in code that needs to run in prod. Instead, this has to stay as the if statements we had.